presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/plancheckerprovidermanagers/PlanCheckerProviderManager.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity.plancheckerprovidermanagers;
+
+import com.facebook.presto.spi.plan.PlanCheckerProvider;
+import com.facebook.presto.spi.plan.PlanCheckerProviderFactory;
+import com.facebook.presto.spi.plan.SimplePlanFragmentSerde;
+import com.google.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static java.util.Objects.requireNonNull;
+
+public class PlanCheckerProviderManager
+{
+    private static final File PLAN_CHECKER_PROVIDER_CONFIG = new File("etc/plan-checker-provider.properties");
+    private final SimplePlanFragmentSerde simplePlanFragmentSerde;
+    private PlanCheckerProviderFactory providerFactory;
+    private PlanCheckerProvider provider;
+
+    @Inject
+    public PlanCheckerProviderManager(SimplePlanFragmentSerde simplePlanFragmentSerde)
+    {
+        this.simplePlanFragmentSerde = requireNonNull(simplePlanFragmentSerde, "planNodeSerde is null");
+        this.provider = new EmptyPlanCheckerProvider();
+    }
+
+    public void addPlanCheckerProviderFactory(PlanCheckerProviderFactory providerFactory)
+    {
+        if (this.providerFactory != null) {
+            throw new IllegalArgumentException("PlanCheckerProviderFactory is already added, only one factory supported");
+        }
+        this.providerFactory = requireNonNull(providerFactory, "providerFactory is null");
+    }
+
+    public PlanCheckerProvider getPlanCheckerProvider()
+    {
+        return provider;
+    }
+
+    public void loadPlanCheckerProvider()
```

Comment:
probably the config will be specific to each plan checker we support (similar to how it works for connectors or function registry), and then adding the configuration file is how you say you want to enable the plugin, and also how you set any configs for it.

Reply:
I thought that specifying the plugin in the coordinator config was to enable it. This will currently load config from "etc/plan-checker-provider.properties" and provide those properties to the loaded plugin. That seemed to be how other plugin providers have the config setup. Are you saying you want `PlanCheckerProviderManager` to check the plan-checker-provider.properties to enable/disable the provided plugin?

Reply:
i think i'm suggesting a variation, where we can have multiple plan checker providers, and then have configs like etc/plan-checker-providers/native-plan-checker.properties to add specific plan checker providers.  That way, for example, i could add the native plan checker, and also some custom plan checkers that are relevant only to my company's needs.

Reply:
Ah yes, I was wondering about having multiple plan checkers, so this sounds good to me. I'll fix this up so that can be supported.

Reply:
@rschlussel just want to make sure I have the semantics of loading multiple plan checker providers right:

1. For each plan checker provider plugin: multiple `PlanCheckerProviderFactory`s can be defined
2. For each `PlanCheckerProviderFactory`: the factory is added to the PlanCheckerProviderManager through `addPlanCheckerProviderFactory`, 
3. Calling `PlanCheckerProviderFactory.create()` the factory is responsible for loading configuration and creating a `PlanCheckerProvider`
4. The provider can use the configuration to determine if specific `PlanChecker`s are enabled/disabled etc.

Sound ok?

Reply:
This seems reasonable to me.